internal/olm/operator/manager.go

```diff
@@ -0,0 +1,376 @@
+	"fmt"
+	"io/ioutil"
+
+	olmresourceclient "github.com/operator-framework/operator-sdk/internal/olm/client"
+	opinternal "github.com/operator-framework/operator-sdk/internal/olm/operator/internal"
+	"github.com/operator-framework/operator-sdk/internal/util/k8sutil"
+	registryutil "github.com/operator-framework/operator-sdk/internal/util/operator-registry"
+	"github.com/operator-framework/operator-sdk/internal/util/yamlutil"
+
+	olmapiv1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1"
+	olmapiv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	apiextv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// TODO(estroz): ensure OLM errors are percolated up to the user.
+
+var (
+	Scheme = olmresourceclient.Scheme
+
+	defaultNamespace = "default"
+)
+
+func init() {
+	if err := apiextv1beta1.AddToScheme(Scheme); err != nil {
+		log.Fatalf("Failed to add Kubhernetes API extensions v1beta1 types to scheme: %v", err)
+	}
+	if err := olmapiv1.AddToScheme(Scheme); err != nil {
+		log.Fatalf("Failed to add OLM operator API v1 types to scheme: %v", err)
+	}
+}
+
+type operatorManager struct {
+	client    *olmresourceclient.Client
+	version   string
+	namespace string
+	force     bool
+
+	installMode           olmapiv1alpha1.InstallModeType
+	installModeNamespaces []string
+	olmObjects            []runtime.Object
+	manifests             registryutil.ManifestsStore
+}
+
+func (c *OLMCmd) newManager() (*operatorManager, error) {
+	m := &operatorManager{
+		force:   c.Force,
+		version: c.OperatorVersion,
+	}
+	rc, ns, err := k8sutil.GetKubeconfigAndNamespace(c.KubeconfigPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get namespace from kubeconfig %s", c.KubeconfigPath)
+	}
+	if ns == "" {
+		ns = defaultNamespace
+	}
+	if c.OperatorNamespace == "" {
+		m.namespace = ns
+	} else {
+		m.namespace = c.OperatorNamespace
+	}
+	if m.client == nil {
+		m.client, err = olmresourceclient.ClientForConfig(rc)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create SDK OLM client")
+		}
+	}
+	for _, path := range c.IncludePaths {
+		if path != "" {
+			objs, err := readObjectsFromFile(path)
+			if err != nil {
+				return nil, err
+			}
+			for _, obj := range objs {
+				m.olmObjects = append(m.olmObjects, obj)
+			}
+		}
+	}
+	// Since a Subscription refers to a CatalogSource, supplying one but
+	// not the other is an error.
+	hasSub, hasCatSrc := m.hasSubscription(), m.hasCatalogSource()
+	if hasSub || hasCatSrc && !(hasSub && hasCatSrc) {
+		return nil, errors.New("both a CatalogSource and Subscription must be supplied if one is supplied")
```

Comment:
We can likely relax this requirement in the future, as we can accept an external `CatalogSource` and write a `Subscription` internally referencing that `CatalogSource`.